### ISSUE TITLE
Exponential and logarithmic slider scales not working in Qt applications

### ISSUE THREAD

**Issue by dariosanfilippo** [grame-cncm/faust#547]
Hello, there.

The nonlinear scales do not seem to work in caqt or jaqt applications, whereas they appear to work correctly in the web IDE. See an example code below.

Ciao,
Dario

import("stdfaust.lib");
lin0 = hslider("[0]linear 0-1", .5, 0, 1, .000001);
lin1 = hslider("[3]linear 1-1000", 500, 1, 1000, .000001);
exp0 = hslider("[1]exponential 0-1[scale:exp]", .5, 0, 1, .000001);
exp1 = hslider("[4]exponential 1-1000[scale:exp]", 500, 1, 1000, .000001);
log0 = hslider("[2]logarithmic 0-1[scale:log]", .5, 0, 1, .000001);
log1 = hslider("[5]logarithmic 1-1000[scale:log]", 500, 1, 1000, .000001);
process = lin0 , lin1 , exp0 , exp1 , log0 , log1;

**Comment by SaphiraKai** [comment]
Still experiencing this issue!

**Comment by orlarey** [comment]
I think there's always been a bit of misunderstanding as to how a Faust slider with a logarithmic scale works. So I'll try to explain it as clearly as possible. Imagine that the slider hovers over values arranged on a logarithmic scale instead of a linear scale. This means that when hslider(“freq[scale:log]”, 440,20,20000,1) is in the middle of its stroke, its value is 632. How is this value obtained? Since we're in the middle of the slider's travel over a logarithmic scale, we take the mean value between log(20) and log(20000), then convert it back with the exp function, which gives exp((log(20)+log(20000))/2) = 632. This is why it makes no sense to have a logarithmic scale for a slider whose minimum value is 0 or negative like in Dario's example log0 = hslider("[2]logarithmic 0-1[scale:log]", .5, 0, 1, .000001);

**Comment by SaphiraKai** [comment]
That's a good thing to know that I'll keep in my head, but the problem I'm dealing with is specifically with Qt. All sliders seem to just be treated as linear, I'll attach a comparison between faustide's UI vs Qt:

![Image](https://github.com/user-attachments/assets/35928502-b2cb-4fba-96de-a3992a18c5cd)
![Image](https://github.com/user-attachments/assets/00efc3d9-d12e-4862-9be7-308e0cedf91f)

**Comment by orlarey** [comment]
Here is what I got (caqt). What is your platform ?

<img width="109" alt="Image" src="https://github.com/user-attachments/assets/b84648f6-f4bd-42ff-824e-607102ff8a35" />

```
import("stdfaust.lib");

process = os.osc(freq) * gain <: _,_
	with {
		freq = vslider("Freq[style:knob][scale:log]", 440, 20, 20000, 0.01);
		gain = vslider("Gain[style:knob]", 0.5, 0, 1, 0.01);
	};
```

**Comment by SaphiraKai** [comment]
Linux, using the `lv2-64bits-qt5` architecture in faustide.

Audacity doesn't use the Qt interface at all *but is logarithmic*, Carla uses the Qt UI but is linear, and Bitwig doesn't support LV2 directly but if I load it indirectly through Ildaeil, I get the Qt interface in linear scale.

**Comment by orlarey** [comment]
The problem seems specific to the LV2+QT combination, as if the log scale metadata is being ignored. Let's ask Stéphane @sletz

**Comment by sletz** [comment]
Yes, the code written by @agraef (https://github.com/agraef) does not use scale conversion code that is written [here](https://github.com/grame-cncm/faust/blob/master-dev/architecture/faust/gui/ValueConverter.h). A use-case is done in [QTGUI.h](https://github.com/grame-cncm/faust/blob/master-dev/architecture/faust/gui/QTUI.h) and could be taken as a starting point.
